### PR TITLE
Fix ordering of MappingMetaData

### PR DESCRIPTION
**Pull request by mkleen**
## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [X] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [X] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by seut** [comment]
Can you please highlight what this is for in the commit message and PR title/description?

File: sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java [hunk 1]
```diff
@@ -66,7 +66,7 @@ private IndexMetaData purgeDynamicStringTemplate(IndexMetaData indexMetaData) {
     @VisibleForTesting
     MappingMetaData purgeDynamicStringTemplate(MappingMetaData mappingMetaData, String indexName) {
         Map<String, Object> oldMapping = mappingMetaData.getSourceAsMap();
-        HashMap<String, Object> newMapping = new HashMap<>(oldMapping.size());
+        Map<String, Object> newMapping = new LinkedHashMap<>(oldMapping.size());
```
Comment (seut) [inline sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java]: minor nitpick: could be defined of `HashMap` as we know its type (no need to just expect an interface)

Comment (seut) [reply sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java]: If this won't be backported, you could also use `var` :-)

Comment (mkleen) [reply sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java]: I thought its best practice in java to program against interfaces.

Comment (seut) [reply sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java]: If the var is exposed and only the behaviour of an interface is required, sure.

Comment (mkleen) [reply sql/src/main/java/io/crate/metadata/upgrade/MetaDataIndexUpgrader.java]: How about "var newMapping" ? :-)

**Pull request closed without merging**